code/game/objects/items/weapons/dice.dm

```diff
@@ -16,17 +16,13 @@
 	icon_state = "d2020"
 	sides = 20
 
-/mob/living/carbon/human/throw_item(atom/target)
-	var/atom/movable/item = src.get_active_hand()
-	if (istype(item, /obj/item/weapon/dice/))
-		var/obj/item/weapon/dice/I = item
-		var/result = rand(1, I.sides)
-		var/comment = ""
-		if(I.sides == 20 && result == 20)
-			comment = "Nat 20!"
-		else if(I.sides == 20 && result == 1)
-			comment = "Ouch, bad luck."
-		I.icon_state = "[I.name][result]"
-		..()
-		usr.visible_message("<span class='notice'>\The [I.name] lands on [result]. [comment]</span>")
-	else ..()
\ No newline at end of file
+/obj/item/weapon/dice/throw_impact(atom/hit_atom)
+	..()
+	var/result = rand(1, sides)
+	var/comment = ""
+	if(sides == 20 && result == 20)
+		comment = "Nat 20!"
+	else if(sides == 20 && result == 1)
+		comment = "Ouch, bad luck."
+	icon_state = "[name][result]"
+	usr.visible_message("<span class='notice'>\The [name] lands on [result]. [comment]</span>")

```

Comment:
Using usr is a bad practice. It's unsecure, and can theoretically be hijacked. Further, you have no control over it. **Even further**, there's no need to use the user as the source of this message, as the message concerns the dice themselves. Ergo, use `src`.

Quoting DM Reference here directly:
> **A good rule of thumb is to never put usr in a proc.** Typically this is an unsafe programming practice, even when you know that a certain proc is only ever called by a verb, because it is common during development to change the way or time at which a proc is called. If src would not be the correct choice, it is better to send another argument to your proc with the information it needs.

You can read more here: http://www.byond.com/docs/ref/info.html#/proc/var/usr